### PR TITLE
Extract executor into fuel-core-executor crate

### PR DESCRIPTION
**Pull request by Salka1988**
This PR moves the existing executor module into this separate crate.

### CODE REVIEW

File: crates/database/src/lib.rs [hunk 1]
```diff
@@ -6,9 +6,12 @@
 
 #![deny(clippy::arithmetic_side_effects)]
 #![deny(clippy::cast_possible_truncation)]
-#![deny(missing_docs)]
+// #![deny(missing_docs)]
 #![deny(unused_crate_dependencies)]
 #![deny(warnings)]
+#![allow(unused_variables)]
```
Comment (xgreenx) [inline crates/database/src/lib.rs:12]: fn deploy_contract_with_id(

File: crates/database/src/vm_database.rs [hunk 2]
```diff
@@ -0,0 +1,336 @@
+use anyhow::anyhow;
```
Comment (xgreenx) [inline crates/database/src/vm_database.rs:1]: I double-checked, the logic is the same as before=)

File: crates/fuel-core/proptest-regressions/query/subscriptions/test.txt [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Seeds for failure cases proptest has generated in the past. It is
```
Comment (xgreenx) [inline crates/fuel-core/proptest-regressions/query/subscriptions/test.txt:1]: Why do we need this file?=)

File: crates/storage/Cargo.toml [hunk 4]
```diff
@@ -22,6 +22,7 @@ derive_more = { workspace = true }
 fuel-core-types = { workspace = true, default-features = false }
 fuel-vm-private = { workspace = true, default-features = false }
 mockall = { workspace = true, optional = true }
+serde = { workspace = true, features = ["derive"] }
```
Comment (xgreenx) [inline crates/storage/Cargo.toml:25]: Why do we need `serder`?=)

File: crates/services/executor/src/refs/database.rs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    },
+    transactional::Transactional,
+    Error as StorageError,
+    MerkleRootStorage,
+    StorageInspect,
+    StorageMutate,
+};
+use fuel_core_types::fuel_tx::ContractId;
+
+pub trait ExecutorDatabaseTrait<D>:
```
Comment (xgreenx) [inline crates/services/executor/src/refs/database.rs:27]: It should be inside of `ports` file

File: crates/services/executor/src/lib.rs [hunk 6]
```diff
@@ -1,10 +1,12 @@
 #![deny(clippy::arithmetic_side_effects)]
 #![deny(clippy::cast_possible_truncation)]
-#![deny(unused_crate_dependencies)]
+// #![deny(unused_crate_dependencies)]
```
Comment (xgreenx) [inline crates/services/executor/src/lib.rs:3]: We need to fix it=)

File: crates/fuel-core/proptest-regressions/query/subscriptions/test.txt (hunk 3 above)
Comment (Salka1988) [reply crates/fuel-core/proptest-regressions/query/subscriptions/test.txt:1]: should I put this to gitignore?

**Review by xgreenx — APPROVED** [review]
Thank you for colossal work!=)

Could you create an issue to move unit tests to `fuel-core-executor`, please?=)

**Comment by Salka1988** [comment]
> Thank you for colossal work!=)
> 
> Could you create an issue to move unit tests to `fuel-core-executor`, please?=)

Thank you for your availability and valuable advices. I make the issue.

File: crates/database/src/lib.rs [hunk 7]
```diff
@@ -9,6 +9,7 @@
 #![deny(missing_docs)]
 #![deny(unused_crate_dependencies)]
 #![deny(warnings)]
+#![deny(unused_variables)]
```
Comment (Dentosal) [inline crates/database/src/lib.rs:12]: nit: isn't this implied by `#![deny(warnings)]`?

**Pull request merged**